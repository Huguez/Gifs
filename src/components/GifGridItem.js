import React from 'react'
import { PropTypes } from 'prop-types';

export const GifGridItem = ( {imagen, title, id} ) => {

    // console.log( imagen );
    return (
        <div className="card animate__animated animate__fadeIn">
            <p> { title } </p>
            <img alt={title} src={ imagen } />
        </div>
    )
}

GifGridItem.prototype = {
    imagen: PropTypes.string.isRequired,
    title:  PropTypes.string.isRequired,
    id:     PropTypes.string
}
