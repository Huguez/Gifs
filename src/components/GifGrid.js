import React from "react";
import { GifGridItem } from "./GifGridItem";
import { useFetchGifs } from "../hooks/useFetchGifs";
import Proptypes from 'prop-types';

export const GifGrid = ({ category }) => {

    const { data, loading } = useFetchGifs( category );
    
    return ( <>
        <h3> { category } </h3>
        { loading && <p className="animate__animated animate__flash "> Loading... </p> }

        <div className="card-grid"> 
            { data.map( ( item ) => { 
                return <GifGridItem key={item.id} {...item } />
            } ) }
        </div> 
    </>);
}

GifGrid.prototype  = {
    category: Proptypes.string.isRequired
}