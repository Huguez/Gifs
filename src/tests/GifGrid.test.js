import React from "react";
import "@testing-library/jest-dom";

import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { shallow, configure } from 'enzyme';
import { GifGrid } from "../components/GifGrid";
import { useFetchGifs } from "../hooks/useFetchGifs";
jest.mock( "../hooks/useFetchGifs" );


configure({adapter: new Adapter()});
describe('Pruebas con GifGrid.js', () => {
    const categoria = "tacos"

    test('Deberia hacer match con el snapshot', () => {
        useFetchGifs.mockReturnValue({ data: [], loading: true });

        let wrapper = shallow( <GifGrid category={ categoria } /> )
        expect( wrapper ).toMatchSnapshot();
    })
    

    test('Debe de mostar items cuando se carga imagenes useFetchGifs ', () => {

        const gifs = [{ id: "123", imagen: "https://www.hackeo.com", title: "hackeo" }]

        useFetchGifs.mockReturnValue({ data: gifs, loading: false });

        const wrapper = shallow( <GifGrid category={ categoria } /> )

        expect( wrapper ).toMatchSnapshot();
        
        expect( wrapper.find( 'p' ).exists() ).toBe(false);
        
        expect( wrapper.find('GifGridItem').length ).toBeGreaterThanOrEqual( gifs.length );
    });
    
    test('Debe ', () => {
        
    })
    

});


