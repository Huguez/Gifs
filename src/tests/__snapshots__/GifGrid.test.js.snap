// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas con GifGrid.js Debe de mostar items cuando se carga imagenes useFetchGifs  1`] = `
<Fragment>
  <h3>
     
    tacos
     
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="123"
      imagen="https://www.hackeo.com"
      key="123"
      title="hackeo"
    />
  </div>
</Fragment>
`;

exports[`Pruebas con GifGrid.js Deberia hacer match con el snapshot 1`] = `
<Fragment>
  <h3>
     
    tacos
     
  </h3>
  <p
    className="animate__animated animate__flash "
  >
     Loading... 
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
